src/scenic/core/dynamics.py

```diff
@@ -729,11 +718,13 @@ def __str__(self):
         return f'violated {self.violationType} of {self.behaviorName} on line {self.lineno}'
 
 class PreconditionViolation(GuardViolation):
-    """Raised when a precondition is violated when invoking a behavior."""
+    """Raised when a precondition is violated when invoking a behavior
+    or when a precondition encounters a `RejectionException`."""
     violationType = 'precondition'
 
 class InvariantViolation(GuardViolation):
-    """Raised when an invariant is violated when invoking/resuming a behavior."""
+    """Raised when an invariant is violated when invoking/resuming a behavior
+    or when an invariant encounters a `RejectionException`."""
```

Comment:
These docstrings need to have a 1-line summary (under 90 chars); you can give further details in the body of the docstring after a blank line.